File: oauth_serve/src/main/kotlin/team/jtq/auth/oauth_serve/tools/Result.java
package team.jtq.auth.oauth_serve.tools;

import lombok.Data;
import team.jtq.auth.oauth_serve.entity.ResultStatusCode;


import java.io.Serializable;

/**
 * 接口返回数据格式
 */
@Data
public class Result implements Serializable {

    private static final long serialVersionUID = 1L;
    private boolean success = true;

    public void setSuccess(boolean success) {
        this.success = success;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public Integer getCode() {
        return code;
    }

    public void setCode(Integer code) {
        this.code = code;
    }

    public Object getData() {
        return data;
    }

    public void setData(Object data) {
        this.data = data;
    }

    private String msg = "ok";

    private Integer code = 0;

    private Object data;


    public Result() {

    }
    public Result(int code, String msg, Object data) {
        this.code = code;
        this.msg = msg;
        this.data = data;
    }

    public Result(ResultStatusCode resultStatusCode, Object data) {
        this(resultStatusCode.getCode(), resultStatusCode.getMsg(), data);
    }

    public Result(int code, String msg) {
        this(code, msg, null);
    }

    public Result(ResultStatusCode resultStatusCode) {
        this(resultStatusCode, null);
    }

    public boolean isSuccess() {
        return this.code == 0;
    }

    public static Result ok() {
        return new Result(ResultStatusCode.OK);
    }

    public static Result ok(Object obj) {
        return new Result(ResultStatusCode.OK, obj);
    }

    public static Result error(ResultStatusCode resultStatusCode){
        return new Result(resultStatusCode);
    }

    public static Result error(ResultStatusCode resultStatusCode, Object obj){
        return new Result(resultStatusCode, obj);
    }

    public static Result error(String msg){
        return new Result(5000, msg, null);
    }
}

File: oauth_serve/src/main/kotlin/team/jtq/auth/oauth_serve/tools/Result.kt
package team.jtq.auth.oauth_serve.tools

import lombok.Data
import team.jtq.auth.oauth_serve.entity.ResultStatusCode

import java.io.Serializable


@Data
class Result : Serializable {
    val success: Boolean
        get() = this.code == 0
    private var msg = "ok"

    private var code = 0

    private var data: Any? = null
    constructor() {}

    @JvmOverloads
    constructor(code: Int, msg: String, data: Any? = null) {
        this.code = code
        this.msg = msg
        this.data = data
    }

    @JvmOverloads
    constructor(resultStatusCode: ResultStatusCode, data: Any? = null) : this(
        resultStatusCode.code,
        resultStatusCode.msg,
        data
    ) {
    }

    companion object {
        private const val serialVersionUID = 1L
        @JvmStatic
        fun ok(): Result {
            return Result(ResultStatusCode.OK)
        }
        @JvmStatic
        fun ok(obj: Any?): Result {
            return Result(ResultStatusCode.OK, obj)
        }
        @JvmStatic
        fun error(resultStatusCode: ResultStatusCode): Result {
            return Result(resultStatusCode)
        }
        @JvmStatic
        fun error(resultStatusCode: ResultStatusCode, obj: Any?): Result {
            return Result(resultStatusCode, obj)
        }
        @JvmStatic
        fun error(msg: String): Result {
            return Result(5000, msg, null)
        }
    }
}